test/integration/SessionCacheServiceSpec.scala

```diff
@@ -383,7 +389,12 @@ class SessionCacheSpec extends IntegrationTestsSpec with MockitoSugar with Mongo
 
       val cache = await(sessionCache.cacheRepo.findById(request))
 
-      val expectedJson = toJson(CachedData(subDetails = Some(subscriptionDetails)))
+      val expectedJson = toJson(
+        CachedData(
+          subDetails = Some(subscriptionDetails),
+          submissionCompleteDetails = Some(SubmissionCompleteData(Some(subscriptionDetails), None))
+        )
+      )
```

Comment:
Similar to the above, can we also get a test for when the processing date is already populated to prove that our code retains the existing processing date and only overwrites the subscription details?